sdk/app_cpu1/common/drv/timing_manager.c

```diff
@@ -208,16 +213,75 @@ double timing_manager_get_tick_delta(void)
 }
 
 /*
- * Set the user-defined ratio for the event qualifier
+ * Gets the expected time (in us) between ISR calls based on the user's switching frequency,
+ * PWM event sub-ratio, and enabled sensors
+ */
+double timing_manager_expected_tick_delta(void)
+{
+    double fsw = pwm_get_switching_freq();
+
+    // The no-sensor-time is the expected time in us between scheduler interrupts
+    // with no sensors enabled. This is just the switching period multiplied by the sub-ratio
+    double no_sensor_time = (now_ratio / fsw) * 1e6;
+
+    // It is possible that the user could ask for a very low sub-ratio, and the sensors with
+    // a longer acquistion time may run over the no-sensor-time, which will cause the sensor
+    // triggering and ISR call to occur at the first valid multiple of the no-sensor-time instead
+    //
+    // Therefore, we should check all the sensors in order of longest to shortest acquistion to see
+    // if the longest enabled sensor will overrun the time calculated above
+    uint16_t enabled_sensors = Xil_In16(TM_BASE_ADDR + TM_SENSOR_EN_CFG_REG_OFFSET);
+    bool is_amds_enabled = enabled_sensors & 0x3C;
+    bool is_eddy_enabled = enabled_sensors & 0x3C0;
+    bool is_adc_enabled = enabled_sensors & 0x1;
+    bool is_encoder_enabled = enabled_sensors & 0x2;
```

Comment:
Move these checks to a public macro function:

`#define IS_ANY_AMDS_ENABLED(en_bits) (en_bits & 0x3C)`

this way the check is abstracted away to make the code clearer, and this can be used by any code, not just here.